src/app/routes/accountRecoveryRoute.js

```diff
@@ -6,76 +6,110 @@ const uuidv1 = require('uuid/v1')
 const proxy = require('express-http-proxy')
 const proxyUtils = require('../proxy/proxyUtils.js')
 const logger = require('sb_logger_util_v2');
-const {encrypt} = require('../helpers/crypto');
+const { encrypt, decrypt } = require('../helpers/crypto');
+const { parseJson, isDateExpired } = require('../helpers/utilityService');
 
 module.exports = (app) => {
 
   app.post('/learner/user/v1/fuzzy/search', proxy(envHelper.learner_Service_Local_BaseUrl, {
     proxyReqOptDecorator: proxyUtils.decorateRequestHeaders(),
     proxyReqPathResolver: (req) => {
-      logger.info({msg: '/learner/user/v1/fuzzy/search called'});
+      logger.info({ msg: '/learner/user/v1/fuzzy/search called' });
       return '/private/user/v1/search';
     }
   }))
 
-  app.post('/learner/user/v1/password/reset', bodyParser.urlencoded({ extended: false }), bodyParser.json({ limit: '10mb' }), 
+  app.post('/learner/user/v1/password/reset', bodyParser.urlencoded({ extended: false }), bodyParser.json({ limit: '10mb' }),
     (req, res, next) => {
-      logger.info({msg: '/learner/user/v1/password/reset called'});
-      if(_.get(req.body, 'request.userId') !== _.get(req.session, 'otpVerifiedFor.request.userId')){
+      logger.info({ msg: '/learner/user/v1/password/reset called' });
+      try {
+        var reqUserId = _.get(req.body, 'request.userId')
+        var reqValidator = _.get(req, 'body.request.validator')
+        var decodedValidator = isValidRequest(reqValidator);
+        // checking only for the userID from request and from the decoded object.
+        if ((decodedValidator['userId']) && (reqUserId === decodedValidator['userId'])) {
+          next();
+        } else {
+          logger.error({
+            msg: 'unauthorized'
+          });
+          res.status(401).send({ "id": "api.reset.password", "ver": "v1", "ts": dateFormat(new Date(), 'yyyy-mm-dd HH:MM:ss:lo'), "params": { "resmsgid": null, "msgid": uuidv1(), "err": null, "status": "unauthorized", "errmsg": null }, "responseCode": "UNAUTHORIZED", "result": { "response": "unauthorized" } })
```

Comment:
Can you add logger info with 
UserId, 
msg: "portal - Unauthorised: Reset password failed"
req.body